server/src/main/java/org/apache/cloudstack/backup/BackupManagerImpl.java

```diff
@@ -316,8 +316,8 @@ public boolean restoreVMBackup(final Long backupId, final String restorePointId)
             throw new CloudRuntimeException("VM " + vmId + " couldn't be found on existing or removed VMs");
         }
         accountManager.checkAccess(CallContext.current().getCallingAccount(), null, true, vm);
-        if (vm.getRemoved() == null && (!vm.getState().equals(VirtualMachine.State.Stopped) ||
-                !vm.getState().equals(VirtualMachine.State.Destroyed))) {
+        if (vm.getRemoved() == null && !vm.getState().equals(VirtualMachine.State.Stopped) &&
```

Comment:
:) better extract this condition as method to make it more readable. chances are it will be inlined during optimisation anyway.

Reply:
Sorry just saw your comment @DaanHoogland. Let's push it as part of the optimisation we are doing in the next phase